File: Source/AlphaTab/Audio/Synth/Midi/Event/MetaDataEvent.cs
using AlphaTab.IO;

namespace AlphaTab.Audio.Synth.Midi.Event
{
    internal class MetaDataEvent : MetaEvent
    {
        public byte[] Data { get; private set; }

        public MetaDataEvent(int delta, byte status, byte metaId, byte[] data)
            : base(delta, status, metaId, 0)
        {
            Data = data;
        }

        public override void WriteTo(IWriteable s)
        {
            s.WriteByte(0xFF);
            s.WriteByte((byte)MetaStatus);

            var l = Data.Length;
            MidiFile.WriteVariableInt(s, l);
            s.Write(Data, 0, Data.Length);
        }
    }
}

File: src/midi/MetaDataEvent.ts
import { MetaEvent } from '@src/midi/MetaEvent';
import { MidiFile } from '@src/midi/MidiFile';
import { IWriteable } from '@src/io/IWriteable';

export class MetaDataEvent extends MetaEvent {
    public data: Uint8Array;

    public constructor(track:number, delta: number, status: number, metaId: number, data: Uint8Array) {
        super(track, delta, status, metaId, 0);
        this.data = data;
    }

    public writeTo(s: IWriteable): void {
        s.writeByte(0xff);
        s.writeByte(this.metaStatus as number);
        let l: number = this.data.length;
        MidiFile.writeVariableInt(s, l);
        s.write(this.data, 0, this.data.length);
    }
}
